### PR TITLE
feat(eslint-plugin): Add testing-library and jsx-a11y rules.

### PR DESCRIPTION
**Pull request by doniyor2109**
Most jsx-a11y rules are take from  [eslint-config-react-app](https://github.com/facebook/create-react-app/blob/master/packages/eslint-config-react-app/index.js)

### CODE REVIEW

**Comment by umidbekkarimov** [comment]
Suggestions for `eslint-plugin-jsx-a11y`

accessible-emoji: Enforce emojis are wrapped in <span> and provide screenreader access.
aria-activedescendant-has-tabindex: Enforce elements with aria-activedescendant are tabbable.
aria-props: Enforce all aria-* props are valid.
aria-proptypes: Enforce ARIA state and property values are valid.
aria-role: Enforce that elements with ARIA roles must use a valid, non-abstract ARIA role.
no-access-key: Enforce that the accessKey prop is not used on any element to avoid complications with 
keyboard commands used by a screenreader.
no-autofocus: Enforce autoFocus prop is not used.
no-redundant-roles: Enforce explicit role property is not the same as implicit/default role property on element.
role-has-required-aria-props: Enforce that elements with ARIA roles must have all required attributes for that role.
role-supports-aria-props: Enforce that elements with explicit or implicit roles defined contain only aria-* properties supported by that role.
scope: Enforce scope prop is only used on <th> elements.
tabindex-no-positive: Enforce tabIndex value is not greater than zero.

File: packages/eslint-plugin/configs/jest.js [hunk 1]
```diff
@@ -48,5 +52,15 @@ module.exports = {
...
+
+    //
+    // eslint-plugin-testing-library
+    //
+    /**
+     * Prevents using `debug` function.
+     *
+     * @see https://github.com/testing-library/eslint-plugin-testing-library/blob/master/docs/rules/no-debug.md
+     * */
+    'testing-library/no-debug': INCONSISTENCY,
```
[the remark this answers is not in the document]
Comment (doniyor2109) [reply packages/eslint-plugin/configs/jest.js]: @umidbekkarimov how about we allow `INCONSISTENCY` for `NODE_ENV=test`?

[the remark this answers is not in the document]
Comment (umidbekkarimov) [reply packages/eslint-plugin/configs/jest.js]: @doniyor2109 I think you don't understand why we use `INCONSISTENCY`, or how it works. `NODE_ENV=development`

File: .eslintrc.js [hunk 2]
```diff
@@ -1,6 +1,29 @@
...
+            message:
+              "We shouldn't use warnings in the base config, we should override errors to warnings in app configs. It's better to use INCONSISTENCY.",
+          },
+        ],
+      },
+    ],
+    'no-restricted-syntax': [
+      'error',
+      {
+        selector: 'Literal[value=warn]',
```
Comment (umidbekkarimov) [inline .eslintrc.js]: there is also number annotations for the rule severity (https://eslint.org/docs/user-guide/configuring#configuring-rules). I think we can't enforce this rule with simple selector

**Pull request merged**